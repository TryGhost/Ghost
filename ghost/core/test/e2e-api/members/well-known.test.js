const {agentProvider, matchers, mockManager} = require('../../utils/e2e-framework');
const {anyString, anyEtag} = matchers;

describe('Members .well-known', function () {
    let membersAgent;

    before(async function () {
        const agents = await agentProvider.getAgentsForMembers();
        membersAgent = agents.membersAgent;
        mockManager.mockMail();
    });

    after(function () {
        mockManager.restore();
    });

    describe('GET /jwks.json', function () {
        it('should return a JWKS', async function () {
            await membersAgent
                .get('/.well-known/jwks.json')
                .expectStatus(200)
                .matchBodySnapshot({
                    keys: [{
                        kid: anyString,
                        n: anyString
                    }]
                })
                .matchHeaderSnapshot({
                    etag: anyEtag
                });
        });
    });
});
