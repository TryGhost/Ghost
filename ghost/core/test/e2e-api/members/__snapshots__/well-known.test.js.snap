// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Members .well-known GET /jwks.json should return a JWKS 1: [body] 1`] = `
Object {
  "keys": Array [
    Object {
      "e": "AQAB",
      "kid": Any<String>,
      "kty": "RSA",
      "n": Any<String>,
    },
  ],
}
`;

exports[`Members .well-known GET /jwks.json should return a JWKS 2: [headers] 1`] = `
Object {
  "access-control-allow-credentials": "true",
  "access-control-allow-origin": "http://127.0.0.1:2369",
  "cache-control": "public, max-age=86400",
  "content-length": "265",
  "content-type": "application/json; charset=utf-8",
  "etag": StringMatching /\\(\\?:W\\\\/\\)\\?"\\(\\?:\\[ !#-\\\\x7E\\\\x80-\\\\xFF\\]\\*\\|\\\\r\\\\n\\[\\\\t \\]\\|\\\\\\\\\\.\\)\\*"/,
  "vary": "Origin, Accept-Encoding",
  "x-powered-by": "Express",
}
`;
